README.md

```diff
@@ -9,20 +9,125 @@ Apache Kafka. It allows you to read/write data from/to Kafka using Numaflow.
 
 ### Source Vertex
 
-TODO
+Kafka Sourcer reads data from a Kafka topic and passes the data to the downstream vertices. Below is an example of how
+to use.
+
+In this example, we read data from a Kafka topic named `numagen` and pass the data to a built-in cat UDF and a built-in
+log sink. Feel free to use your own topic when trying this example.
+
+#### Steps
+
+1. Configure the source vertex in a K8s ConfigMap. The configurations include 2 parts: Kafka consumer properties and
+   user configurations. Fill in the consumer properties according to your Kafka cluster as well as add the
+   topic name and consumer group id in the user configurations.
+
+```yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-config
+data:
+  consumer.properties: |
+    # Required connection configs for Kafka consumer
+    bootstrap.servers=localhost:9092
+    security.protocol=SASL_SSL
+    sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username='user' password='password';
+    sasl.mechanism=PLAIN
+    # Required for correctness in Apache Kafka clients prior to 2.6
+    client.dns.lookup=use_all_dns_ips
+    # Best practice for higher availability in Apache Kafka clients prior to 3.0
+    session.timeout.ms=45000
+    # Best practice for Kafka producer to prevent data loss
+    acks=all
+    # Schema Registry connection configurations
+    schema.registry.url=localhost:8081
+    basic.auth.credentials.source=USER_INFO
+    basic.auth.user.info=a:b
+    # Other configurations
+    retries=0
+    # group.id is required for consumer clients
+    group.id=group1
+  user.configuration: |
+    topicName: numagen
```

Comment:
Do we want to support topic from different brokers? For example.. reading from 2 different regions then consumer properties and topic name will have to go hand in hand.

Reply:
https://github.com/numaproj-contrib/kafka-java/issues/7